ui/src/attestation/contract.ts

```diff
@@ -53,24 +53,45 @@ export class ClaimsContract {
     transaction.add(transaction.claimRadicleIdentity(tx, urn));
   }
 
+  // Fetches the identity claimed by the given Ethereum address.
+  // Returns `undefined` if currently there's no valid claim.
+  async getClaimed(address: string): Promise<Uint8Array | undefined> {
+    const filter = this.contract.filters.Claimed(address);
+    const lastEvent = (await this.contract.queryFilter(filter)).pop();
+    if (!lastEvent) {
+      return undefined;
+    }
+    let claimed;
+    try {
+      claimed = await this.getClaimedByTx(lastEvent.transactionHash, address);
+    } catch {
+      return undefined;
+    }
+    if (claimed.every(hashByte => hashByte === 0)) {
+      return undefined;
```

Comment:
I don’t understand we why are doing this check. Could you add a comment?

Reply:
Added. This branch could be dropped, but then we would be returning a `0` hash if the identity has been explicitly unclaimed. That would needlessly complicate the function's API.